### PR TITLE
Imxrt-1020: Bring up nxp_demo and loadable_elf_apps configs

### PR DESCRIPTION
**Pull request by kishore-sn**
This PR contains following changes:
- Fix dtcm configuration issue for imxrt 1020
- Temporary fix for nxp_demo config boot up issue
- Bring up loadable_elf_apps config

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [66e4bb8ee887fc468e249be13a5a172df7f9005f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [66e4bb8ee887fc468e249be13a5a172df7f9005f] - Code Rule Check OK.

File: build/configs/imxrt1020-evk/scripts/memory.ld [hunk 1]
```diff
@@ -85,17 +85,25 @@
...
+  /* 64MB of HyperFLASH */
 
-  kflash (rx)  : ORIGIN = 0x60000000, LENGTH = 1M
-  uflash (rx)  : ORIGIN = 0x60200000, LENGTH = 1M
+  kflash (rx)  : ORIGIN = 0x60000000, LENGTH = 256K
+  uflash (rx)  : ORIGIN = 0x60040000, LENGTH = 256K
   flash  (rx)  : ORIGIN = 0x60400000, LENGTH = 62M
 
-  /* 512Kb of OCRAM */
+  /* 512KB of OCRAM */
```
[the remark this answers is not in the document]
Comment (kishore-sn) [reply build/configs/imxrt1020-evk/scripts/memory.ld]: ok. fixed.

**Comment by seinfra** [comment]
Target : [3c0cc2e8b14e9c7e5626714eeee9313d68435d8a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3c0cc2e8b14e9c7e5626714eeee9313d68435d8a] - Code Rule Check OK.

File: build/configs/imxrt1020-evk/scripts/user-space-dtcm.ld [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * build/configs/imxrt1050-evk/scripts/user-space.ld
```
Comment (NilabhShant) [inline build/configs/imxrt1020-evk/scripts/user-space-dtcm.ld]: Please correct the file path.

Comment (kishore-sn) [reply build/configs/imxrt1020-evk/scripts/user-space-dtcm.ld]: fixed.

**Comment by seinfra** [comment]
Target : [132229d528ea3138e98ee89300bc7ee0af1bb1af] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [132229d528ea3138e98ee89300bc7ee0af1bb1af] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9f73a440902fcfa2e1c664c44a9ac3361f34b316] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9f73a440902fcfa2e1c664c44a9ac3361f34b316] - Code Rule Check (C++) OK.

File: build/configs/imxrt1020-evk/scripts/memory.ld [hunk 3]
```diff
@@ -85,17 +85,25 @@
 
 MEMORY
 {
-  /* 64Mb of HyperFLASH */
+  /* 64MB of HyperFLASH */
```
Comment (sunghan-chang) [inline build/configs/imxrt1020-evk/scripts/memory.ld]: It has 8MB nor flash, not 64MB of hyperflash.

**Comment by seinfra** [comment]
Target : [333e81fb84672b1c7e78877a5171e12ddd25f69e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [333e81fb84672b1c7e78877a5171e12ddd25f69e] - Code Rule Check OK.

File: build/configs/imxrt1020-evk/scripts/memory.ld (hunk 3 above)
Comment (kishore-sn) [reply build/configs/imxrt1020-evk/scripts/memory.ld]: Fixed.

File: build/configs/imxrt1020-evk/scripts/memory.ld [hunk 4]
```diff
@@ -85,17 +85,25 @@
...
 MEMORY
 {
-  /* 64Mb of HyperFLASH */
+  /* 8MB of HyperFLASH */
 
-  kflash (rx)  : ORIGIN = 0x60000000, LENGTH = 1M
-  uflash (rx)  : ORIGIN = 0x60200000, LENGTH = 1M
+  kflash (rx)  : ORIGIN = 0x60000000, LENGTH = 256K
+  uflash (rx)  : ORIGIN = 0x60040000, LENGTH = 256K
   flash  (rx)  : ORIGIN = 0x60400000, LENGTH = 62M
```
Comment (sunghan-chang) [inline build/configs/imxrt1020-evk/scripts/memory.ld]: Don't you need to modify LENGTH of flash?

Comment (kishore-sn) [reply build/configs/imxrt1020-evk/scripts/memory.ld]: Actually, that value is never used. So I have not updated it. 
Also, if we need to specify the length, then it has to be the remaining length
which is 8M - 512K = 7.5M, 
and I don't know how to specify this value in the ld script. 

I think its ok to keep it for now. This line will be removed when we move to the auto-generate method for generating ld script from config file.

**Pull request merged**